### DISCUSSION TITLE
how to remove prometheus from Inbound in k8s

### DISCUSSION THREAD

**Discussion by RiverFerry** [linkerd/linkerd2#8854] [Help] [answered]
i have installed linkerd-viz in my k8s cluster, i can view grafana dashboard by viz web, and latency here works great for me， it has defalut config:

```
histogram_quantile(0.5, sum(irate(response_latency_ms_bucket{namespace="$namespace", deployment="$deployment", direction="inbound"}[30s])) by (le, deployment))
```

my problem is, in my deployment A,  inbound has multiple sources，like:
* B
* C
* D
* deploy/prometheus

deploy/prometheus exist in any of deployment, but i wish p50 latency can remove `deploy/prometheus` from inbound, so i can measure my own deployment app. is there has any solution? thanks!

**Comment by mateiidavid** [comment, marked as answer]
You can try to drop metrics with specific labels by modifying the prometheus scrape config.

https://github.com/linkerd/linkerd2/blob/80b2fdbe3f92650414e054ea3d40dbfea2e27111/viz/charts/linkerd-viz/templates/prometheus.yaml#L88

For example, the proxy job in the snippet above will configure prometheus to scrape the proxies' admin port. You might be able to drop labels based on regex using `labeldrop`?

**Comment by RiverFerry** [comment]
thanks for your reply, i will try this.
